src/components/form/fields/FileUploadField.tsx

```diff
@@ -59,14 +53,12 @@ export function FileUploadField({
         };
       }),
     );
-    if (isMultiple) {
-      field.onChange(fileObjects);
-      onChange?.(fileObjects);
-      return;
+    if (multiple) {
+      return fileObjects;
```

Comment:
Perhaps we could simply return fileObjects. Consumers can than check whether the array has multiple files or not. Wdyt? I would also rename `fileObjects` to `base64Files`.